test/unit/test_unit.ml

```diff
@@ -27,6 +27,51 @@ module Test_location = struct
   let tests = test_compare_width_decreasing
 end
 
+module Test_conf = struct
+  let test_check_version =
+    let test name ~expected ~exe res =
+      let test_name = "compare_version: " ^ name in
+      ( test_name
+      , `Quick
+      , fun () ->
+          let got = Conf.check_version ~expected ~exe in
+          Alcotest.check Alcotest.(result unit string) test_name res got )
+    in
+    [ test "empty 'expected'" ~expected:"" ~exe:""
+        (Error "malformed version number \"\".")
+    ; test "invalid 'expected'" ~expected:"0.33..foo" ~exe:""
+        (Error "malformed version number \"0.33..foo\".")
+    ; test "invalid 'exe'" ~expected:"0.14.2" ~exe:"0.33..foo"
+        (Error
+           "expected ocamlformat version to be \"0.14.2\" but got \
+            \"0.33..foo\".")
+    ; test "unknown 'exe'" ~expected:"0.14.2" ~exe:"unknown"
+        (Error
+           "expected ocamlformat version to be \"0.14.2\" but got \
+            \"unknown\".")
+    ; test "equal" ~expected:"0.14.1" ~exe:"0.14.1" (Ok ())
+    ; test "accepted" ~expected:"0.14.1" ~exe:"0.14.1-15-g273f6f6-dirty"
+        (Ok ())
+    ; test "too old" ~expected:"0.14.1" ~exe:"0.14.0"
+        (Error
+           "expected ocamlformat version to be at least \"0.14.1\" but got \
```

Comment:
When receiving this message I would be tempted to upgrade to 0.14.2. "at least" is not true here.
I think the fact that `0.14.0-something` is between `0.14.0` and `0.14.1` is not necessary for users and should be removed.

Reply:
I simplified the error message